src/scipp/core/like.py

```diff
@@ -0,0 +1,128 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2022 Scipp contributors (https://github.com/scipp)
+
+from typing import Any as _Any, Union as _Union
+from .._scipp import core as _cpp
+from .variable import ones, zeros, empty, full
+
+
+def _to_variable_or_data_array(var, new_values):
+    if isinstance(var, _cpp.DataArray):
+        return _cpp.DataArray(data=new_values,
+                              coords={c: coord
+                                      for c, coord in var.coords.items()},
+                              attrs={a: attr
+                                     for a, attr in var.attrs.items()},
+                              masks={m: mask.copy()
+                                     for m, mask in var.masks.items()})
+    else:
+        return new_values
+
+
+def zeros_like(
+    var: _Union[_cpp.Variable,
+                _cpp.DataArray]) -> _Union[_cpp.Variable, _cpp.DataArray]:
+    """
+    If input is a :class:`Variable`, it constructs a :class:`Variable` with the same
+    dims, shape, unit and dtype, but with all values initialized to 0. If the input
+    has variances, all variances in the output are set to 0.
+
+    If input is a :class:`DataArray`, it constructs a :class:`DataArray` with the same
+    coordinates, attributes and masks as the input, but sets the data values to 1.
```

Comment:
0?

Maybe the docstring is more readable if we unify the description? I don't see the need for writing the same thing twice? Something like:

```
Constructs a new object with the same dims, shape, unit and dtype as the input (Variable
or DataArray), but with all values initialized to 0. If the input has variances, all variances
in the output are set to 0.
If the input is a data array, coordinates and attributes are shallow-copied and masks are
deep copied.
```